integration-tests/emulated/tests/bridges/bridge-hub-kusama/src/tests/snowbridge.rs

```diff
@@ -0,0 +1,681 @@
+			<AssetHubKusama as Chain>::RuntimeOrigin::root(),
+			weth_asset_id,
+			asset_hub_sovereign.clone().into(),
+			true,
+			1000,
+		));
+
+		assert!(<AssetHubKusama as AssetHubKusamaPallet>::ForeignAssets::asset_exists(
+			weth_asset_id
+		));
+	});
+
+	BridgeHubKusama::execute_with(|| {
+		type RuntimeEvent = <BridgeHubKusama as Chain>::RuntimeEvent;
+
+		type RuntimeOrigin = <BridgeHubKusama as Chain>::RuntimeOrigin;
+
+		// Fund AssetHub sovereign account so it can pay execution fees for the asset transfer
+		<BridgeHubKusama as BridgeHubKusamaPallet>::Balances::force_set_balance(
+			RuntimeOrigin::root(),
+			asset_hub_sovereign.clone().into(),
+			INITIAL_FUND,
+		)
+		.unwrap();
+
+		let message_id: H256 = [1; 32].into();
+		let message = VersionedMessage::V1(MessageV1 {
+			chain_id: CHAIN_ID,
+			command: Command::SendToken {
+				token: WETH.into(),
+				destination: Destination::ForeignAccountId32 {
+					para_id: PenpalA::para_id().into(),
+					id: PenpalAReceiver::get().into(),
+					fee: 40_000_000_000,
+				},
+				amount: 1_000_000,
+				fee: 40_000_000_000,
+			},
+		});
+		// Convert the message to XCM
+		let (xcm, _) = EthereumInboundQueue::do_convert(message_id, message).unwrap();
+		// Send the XCM
+		let _ = EthereumInboundQueue::send_xcm(xcm, AssetHubKusama::para_id()).unwrap();
+
+		assert_expected_events!(
+			BridgeHubKusama,
+			vec![
+				RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) => {},
+			]
+		);
+	});
+
+	AssetHubKusama::execute_with(|| {
+		type RuntimeEvent = <AssetHubKusama as Chain>::RuntimeEvent;
+		// Check that the assets were issued on AssetHub
+		assert_expected_events!(
+			AssetHubKusama,
+			vec![
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { .. }) => {},
+				RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) => {},
+			]
+		);
+	});
+
+	PenpalA::execute_with(|| {
+		type RuntimeEvent = <PenpalA as Chain>::RuntimeEvent;
+		// Check that the assets were issued on PenPal
+		assert_expected_events!(
+			PenpalA,
+			vec![
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { .. }) => {},
+			]
+		);
+	});
+}
+
+/// Tests the registering of a token as an asset on AssetHub, and then subsequently sending
+/// a token from Ethereum to AssetHub.
+#[test]
+fn send_token_from_ethereum_to_asset_hub() {
+	BridgeHubKusama::fund_para_sovereign(AssetHubKusama::para_id().into(), INITIAL_FUND);
+
+	// Fund ethereum sovereign on AssetHub
+	AssetHubKusama::fund_accounts(vec![(AssetHubKusamaReceiver::get(), INITIAL_FUND)]);
+
+	BridgeHubKusama::execute_with(|| {
+		type RuntimeEvent = <BridgeHubKusama as Chain>::RuntimeEvent;
+
+		// Construct RegisterToken message and sent to inbound queue
+		send_inbound_message(make_register_token_message()).unwrap();
+
+		// Construct SendToken message and sent to inbound queue
+		send_inbound_message(make_send_token_message()).unwrap();
```

Comment:
```suggestion
		// Construct RegisterToken message and sent to inbound queue
		assert_ok!(send_inbound_message(make_register_token_message()));

		// Construct SendToken message and sent to inbound queue
		assert_ok!(send_inbound_message(make_send_token_message()));
```

@claravanstaden please use `asset_ok!` everywhere for `send_inbound_message`

Reply:
Changed in https://github.com/polkadot-fellows/runtimes/pull/130/commits/fdea2730c3a36229f273e6c8f2d0393e03730206.